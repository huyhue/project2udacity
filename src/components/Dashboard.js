import { connect } from "react-redux";
import Question from "./Question";

const Dashboard = ({ authedUser, questions, users }) => {

  const unanswered = (question) =>
    !question.optionOne.votes.includes(authedUser.id) &&
    !question.optionTwo.votes.includes(authedUser.id);

  const answered = (question) =>
    question.optionOne.votes.includes(authedUser.id) ||
    question.optionTwo.votes.includes(authedUser.id);


  return (
    <div>
      <h1 className="text-center m-2" data-testid="dashboard">
        DASHBOARD
      </h1>

      <h2 className="bg-success-subtle"><span>New Questions</span></h2>
      <div className="text-center" data-testid="unanswered">
        {questions.filter(unanswered).length === 0 && <strong className="text-success">All questions are answered.</strong>}
        {questions.filter(unanswered).map((question) => (
            <Question
            key={question.id}
            question={question}
            author={users[question.author]}
            isAnswered={false}
            /> 
        ))}
      </div>

      <hr />
    
      <h2 className="bg-secondary-subtle"><span>Answered Questions</span></h2>
      <div className="text-center" data-testid="answered">
        {questions.filter(answered).length === 0 && <strong className="text-success">No questions are answered.</strong>}
        {questions.filter(answered).map((question) => (
            <Question
            key={question.id}
            question={question}
            author={users[question.author]}
            isAnswered={true}
            />
        ))}
      </div>
    </div>
  );
};

const mapStateToProps = ({ authedUser, questions, users }) => ({
  authedUser,
  questions: Object.values(questions).sort((a, b) => b.timestamp - a.timestamp),
  users,
});

export default connect(mapStateToProps)(Dashboard);