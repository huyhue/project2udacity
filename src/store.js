import reducers from './reducers';
import middleware from "./middlewares";
import { legacy_createStore as createStore } from "redux";

const store = createStore(reducers, middleware);

export default store;
