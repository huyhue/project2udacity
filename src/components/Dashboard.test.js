import { render } from "@testing-library/react";
import '@testing-library/jest-dom'
import React from "react";
import {Provider} from "react-redux";
import store from "../store";
import {BrowserRouter} from "react-router-dom";
import Dashboard from "./Dashboard";

describe("Dashboard", () => {
    it("should render screen", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Dashboard />
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });
})