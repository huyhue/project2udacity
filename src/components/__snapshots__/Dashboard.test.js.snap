// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render screen 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="text-center m-2"
          data-testid="dashboard"
        >
          DASHBOARD
        </h1>
        <h2
          class="bg-success-subtle"
        >
          <span>
            New Questions
          </span>
        </h2>
        <div
          class="text-center"
          data-testid="unanswered"
        >
          <strong
            class="text-success"
          >
            All questions are answered.
          </strong>
        </div>
        <hr />
        <h2
          class="bg-secondary-subtle"
        >
          <span>
            Answered Questions
          </span>
        </h2>
        <div
          class="text-center"
          data-testid="answered"
        >
          <strong
            class="text-success"
          >
            No questions are answered.
          </strong>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="text-center m-2"
        data-testid="dashboard"
      >
        DASHBOARD
      </h1>
      <h2
        class="bg-success-subtle"
      >
        <span>
          New Questions
        </span>
      </h2>
      <div
        class="text-center"
        data-testid="unanswered"
      >
        <strong
          class="text-success"
        >
          All questions are answered.
        </strong>
      </div>
      <hr />
      <h2
        class="bg-secondary-subtle"
      >
        <span>
          Answered Questions
        </span>
      </h2>
      <div
        class="text-center"
        data-testid="answered"
      >
        <strong
          class="text-success"
        >
          No questions are answered.
        </strong>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
